### PR TITLE
Recognize digits new api first draft

### PR DESCRIPTION
**Pull request by jetfuel**
This PR cover a bit of the Chinese version. Please don't mind them. I will focus on the English version first. and port the update to Chinese later.

### CODE REVIEW

File: 02.recognize_digits/README.md [hunk 1]
```diff
@@ -124,183 +136,179 @@ PaddlePaddle provides a Python module, `paddle.dataset.mnist`, which downloads a
 |t10k-labels-idx1-ubyte |  Evaluation labels | 10,000 |
 
 
+## Fluid API Overview
```
Comment (nickyfantasy) [inline 02.recognize_digits/README.md:139]: This is good summarization. Should we include that in Chapter 1? @daming-lu

File: 02.recognize_digits/README.md [hunk 2]
```diff
@@ -124,183 +136,179 @@ PaddlePaddle provides a Python module, `paddle.dataset.mnist`, which downloads a
...
+def softmax_regression():
+    img = fluid.layers.data(name='img', shape=[1, 28, 28], dtype='float32')
     predict = paddle.layer.fc(input=img,
                               size=10,
                               act=paddle.activation.Softmax())
     return predict
 ```
 
-- Multi-Layer Perceptron: this network has two hidden fully-connected layers, one with ReLU and the other with softmax activation:
+- Multi-Layer Perceptron: this network has two hidden fully-connected layers, both are using ReLU as activation functino. The output layer is using softmax activation:
```
Comment (nickyfantasy) [inline 02.recognize_digits/README.md]: functino -> function

File: 02.recognize_digits/README.md [hunk 3]
```diff
@@ -124,183 +136,179 @@ PaddlePaddle provides a Python module, `paddle.dataset.mnist`, which downloads a
...
+### Trainer Configuration
 
-`shuffle` is a reader decorator. It takes a reader A as input and returns a new reader B. Under the hood, B calls A to read data in the following fashion: it copies in `buffer_size` instances at a time into a buffer, shuffles the data, and yields the shuffled instances one at a time. A large buffer size would yield very shuffled data.
+Now, we need to setup the trainer. The trainer need to take in `train_program`, `place`, and `optimizer`.
+In the following `Momentum` optimizer, `momentum=0.9` means that 90% of the current momentum comes from that of the previous iteration. The learning rate relates to the speed at which the network training converges. Regularization is meant to prevent over-fitting; here we use the L2 regularization.
 
-`batch` is a special decorator, which takes a reader and outputs a *batch reader*, which doesn't yield an instance, but a minibatch at a time.
+```python
+ use_cude = False # set to True if training with GPU
```
Comment (nickyfantasy) [inline 02.recognize_digits/README.md]: cude -> cuda

File: 02.recognize_digits/README.md [hunk 4]
```diff
@@ -124,183 +136,179 @@ PaddlePaddle provides a Python module, `paddle.dataset.mnist`, which downloads a
...
-                              act=paddle.activation.Softmax())
-    return predict
+        act="relu")
+    # output layer with softmax activation function. size = 10 since there are only 10 possible digits.
+    prediction = fluid.layers.fc(input=conv_pool_2, size=10, act='softmax')
+    return prediction
 ```
 
-PaddlePaddle provides a special layer `layer.data` for reading data. Let us create a data layer for reading images and connect it to a classification network created using one of above three functions.  We also need a cost layer for training the model.
+#### Train Program Configuration
```
Comment (nickyfantasy) [inline 02.recognize_digits/README.md:223]: Do you intend to make this section a child under Program Functions Configuration? Or it should be just ###

Comment (jetfuel) [reply 02.recognize_digits/README.md:223]: I do feel it should be under the `Program Functions Configuration` since it is a program function. What do you think? @nickyfantasy

File: 02.recognize_digits/README.md (hunk 1 above)
Comment (daming-lu) [reply 02.recognize_digits/README.md:139]: Yes

File: 02.recognize_digits/README.md [hunk 5]
```diff
@@ -124,183 +136,177 @@ PaddlePaddle provides a Python module, `paddle.dataset.mnist`, which downloads a
 |t10k-labels-idx1-ubyte |  Evaluation labels | 10,000 |
 
 
+## Fluid API Overview
+
+The demo will be using the latest paddle fluid API. Fluid API is the latest Paddle API. It simplifies the model configurations without sacrifice the performance.
```
Comment (daming-lu) [inline 02.recognize_digits/README.md:141]: without sacrificing

File: 02.recognize_digits/README.md [hunk 6]
```diff
@@ -124,183 +136,177 @@ PaddlePaddle provides a Python module, `paddle.dataset.mnist`, which downloads a
...
+We recommend using Fluid API as it is much easier to pick up.
+Here are the quick overview on the major fluid API complements.
```
Comment (daming-lu) [inline 02.recognize_digits/README.md:143]: Here is a quick

File: 02.recognize_digits/README.md [hunk 7]
```diff
@@ -124,183 +136,177 @@ PaddlePaddle provides a Python module, `paddle.dataset.mnist`, which downloads a
...
+
+1. `inference_program`: A function that specify how to get the prediction from the data input.
```
Comment (daming-lu) [inline 02.recognize_digits/README.md:145]: specifies

File: 02.recognize_digits/README.md [hunk 8]
```diff
@@ -124,183 +136,177 @@ PaddlePaddle provides a Python module, `paddle.dataset.mnist`, which downloads a
...
+This is where you specify the network flow.
+1. `train_program`: A function that specify how to get avg_cost from `inference_program` and labels.
```
Comment (daming-lu) [inline 02.recognize_digits/README.md:147]: specifies

File: 02.recognize_digits/README.md [hunk 9]
```diff
@@ -315,27 +321,25 @@ Usually, with MNIST data, the softmax regression model achieves an accuracy arou
 
 ## Application
 
-After training, users can use the trained model to classify images. The following code shows how to inference MNIST images through `paddle.infer` interface.
+After training, users can use the trained model to classify images. The following code shows how to inference MNIST images through `fluid.Inferencer`.
```
Comment (daming-lu) [inline 02.recognize_digits/README.md:324]: how to infer or do inference
